go-controller/pkg/ovn/kubevirt.go

```diff
@@ -0,0 +1,88 @@
+package ovn
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	kapi "k8s.io/api/core/v1"
+
+	kvv1 "kubevirt.io/api/core/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kubevirt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func (oc *DefaultNetworkController) ensureDHCPOptionsForVM(pod *corev1.Pod, lsp *nbdb.LogicalSwitchPort) error {
```

Comment:
it's a bit strange to have a pod as the argument for a function named `ensureDHCPOptionsForVM`. 

Could you do it with the namespaced name of the VM instead ? Maybe also the annotations.

Reply:
We need also Annotations to extract PodOvnAnnotation we can change the name to `ensureDHCPOptionsForVMPod`

Reply:
I will remove the namespace, but is the approach ok ? to Extract VM from pods and use it as primary primitive everywhere ?

Reply:
I think its nice to use a neat struct for this. 
I would just refrain from setting the entire pod in it; can't you just grab the relevant things from the pod ? i.e. the namespace, and the relevant annotations ?